CMake/FindArmadillo.cmake

```diff
@@ -91,11 +91,12 @@ if(NOT _ARMA_USE_WRAPPER OR MSVC)
     endif()
   endif()
   if(_ARMA_USE_BLAS)
-    if(ARMADILLO_FIND_QUIETLY OR NOT ARMADILLO_FIND_REQUIRED)
-      find_package(BLAS QUIET)
-    else()
-      find_package(BLAS REQUIRED)
-    endif()
+#    if(NOT BLAS_FOUND)
```

Comment:
Looks like that line can be removed?

Reply:
Just commented out, to see if it can work without.

Reply:
This should be able to work without any changes to `FindArmadillo.cmake`.  I'm sorry to dig in my feet so hard about this issue; basically, I think something is poorly designed if it intertwines two components that don't need to be intertwined.  In this case, the change being made to `FindArmadillo.cmake` is specific to the cross-compilation toolchain... but any CMake `find_package` script is supposed to be generic enough to be used in any situation.  So, if we have code in this generic `find_package` script specific to our cross-compilation setup, we are setting ourselves up for a very difficult debugging time later, since we have introduced some code specific to some exceptional situation that only happens in our cross-compilation setup.  I can virtually guarantee that at some point in the future, that `if (NOT BLAS_FOUND)` condition will get activated in someone else's setup when they are not using cross-compilation, and that will cause all kind of headaches to debug.

Sorry to write so much and belabor the point; I hope my motivations make sense.  I am debugging the issue right now to try and find a solution. :+1:

Reply:
I totally agree, I understand the issue.
I left you a comment above with a link to `Findblas.cmake` to show where the issue is coming from.